ouroboros-network/src/Ouroboros/Network/StrictFingerTree.hs

```diff
@@ -0,0 +1,346 @@
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE CPP                        #-}
+#if __GLASGOW_HASKELL__ >= 706
+{-# LANGUAGE DeriveGeneric              #-}
+#endif
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+
+-- | Strict variants of 'FingerTree' operations.
+--
+-- TODO: Move this to @cardano-prelude@ as a new module, @Data.FingerTree.Strict@.
+module Ouroboros.Network.StrictFingerTree
+  ( FingerTree
+
+    -- * Construction
+  , empty
+  , singleton
+  , (<|)
+  , (|>)
+  , (><)
+  , fromList
+
+    -- * Deconstruction
+  , null
+
+    -- ** Examining the ends
+  , ViewL(..)
+  , ViewR(..)
+  , viewl
+  , viewr
+
+    -- ** Search
+  , SearchResult(..)
+  , search
+
+    -- ** Splitting
+    -- | These functions are special cases of 'search'.
+  , split
+  , takeUntil
+  , dropUntil
+
+    -- * Transformation
+  , reverse
+
+    -- ** Maps
+  , fmap'
+  , fmapWithPos
```

Comment:
Hmm, some of these functions are not used in our codebase. Maybe we can leave them out for now and add them only when they're needed?

Reply:
Will do.